### PR TITLE
kube/configmapwatcher: leverage atomic variable as sync indication

### PR DESCRIPTION
**Pull request by panjf2000**
The `Controller.HasSynced()` is used for `cache.WaitForCacheSync()` which will keep polling until the underlying cache has synced and the callback has been called at least once, thus the synchronization of `hasSynced` mark could be in a more lightweight way: an atomic variable instead of a mutex.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[x] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pkg/kube/configmapwatcher/configmapwatcher.go [hunk 1]
```diff
@@ -44,8 +44,7 @@ type Controller struct {
 	configMapName      string
 	callback           func(*v1.ConfigMap)
 
-	mu        sync.RWMutex
-	hasSynced bool
+	hasSynced atomic.Bool
```
Comment (hzxuzhonghu) [inline pkg/kube/configmapwatcher/configmapwatcher.go:47]: I donot think this is right, the atomic.Bool can not be copied. So this must be a pointer type

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm, but will give others a chance to look

**Review by panjf2000 — COMMENTED** [review]
I know it can't be copied, but Where is this variable copied?

**Comment by panjf2000** [comment]
/ assign @tbarrella

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
I would encourage you to focus on fixing real user issues rather than fixing theoretical performance benefits of functions that are called only a few times and are already fast.

**Comment by panjf2000** [comment]
> I would encourage you to focus on fixing real user issues rather than fixing theoretical performance benefits of functions that are called only a few times and are already fast.

I've been reading the source code of istio recently, and when I saw the code that could be optimized, I opened a PR without much thought, I will be absorbed in what really matters later, thanks for the advices.

**Comment by panjf2000** [comment]
Besides, since you've approved this PR, would you mind removing the `do-not-merge/hold` label?
@howardjohn

**Pull request merged**